src/main/java/com/develhope/imsdevelhope/app/business/services/CustomUserDetailsService.java

```diff
@@ -0,0 +1,23 @@
+package com.develhope.imsdevelhope.app.business.services;
+import com.develhope.imsdevelhope.app.persistence.entities.User;
+import com.develhope.imsdevelhope.app.persistence.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class CustomUserDetailsService implements UserDetailsService {
```

Comment:
you can also mark it with `@Service`. To be identified as a bean hence located by other beans.